geoh5py/objects/cell_object.py

```diff
@@ -69,6 +69,20 @@ def cells(self, cells: np.ndarray | list | tuple):
 
         self.workspace.update_attribute(self, "cells")
 
+    @property
+    def centroids(self) -> np.ndarray | None:
+        """
+        Compute the centroids of the cells.
+        """
+        if self.cells is None:
+            return None
+
+        return np.array([self.vertices[conn].mean(axis=0) for conn in self.cells])
```

Comment:
That's a bit dangerous change since we often check if centroids is None in other repos. Sort out of scope for this issue anyway. Remove unless absolutely necessary

Reply:
ok, but it would be nice to have the centroids of the cells.

If in a curve object, the data are associated to cells, we want the coordinates of thoses cells, so the mean of the vertices delimiting it.

Reply:
Ok we can try. 
Can you turn this operation to be vectorial though? Looping will be very expensive here.